railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -443,6 +443,9 @@ def generate_models_for_animals
         require "#{app_path}/config/environment"
         ActiveRecord::Base.configurations.configs_for(env_name: Rails.env).each do |db_config|
           db_create_and_drop_namespace db_config.name, db_config.database
+        ensure
+          # secondary databases might have been created by check_protected_environments task
+          rails("db:drop:all")
```

Comment:
This change is because the Rake task dependency of `db:drop:primary` on `check_protected_environments` now triggers the creation of the `secondary` database, because well, is now iterating over all configured databases.

One could say that this change is a symptom that the `check_protected_environments` rake task should be split into `check_protected_environments:all` vs `check_protected_environments:#{db_name}` so that db-specific tasks can depend on the db-specific `check_protected_environments:#{db_name}` tasks.

Any thoughts? Do we care?